From: Pnina Feder <pnina.feder@mobileye.com>
Subject: panic: add panic_redirect_cpu declaration to header
Date: Sun, 18 Jan 2026 14:01:49 +0200

Add the extern declaration for panic_redirect_cpu to linux/panic.h to fix
the sparse warning:

  kernel/panic.c:305:10: sparse: sparse: symbol 'panic_redirect_cpu'
  was not declared. Should it be static?

The variable needs to be declared in the header alongside panic_cpu since
it is used for synchronization when redirecting panic to a specific CPU
via the panic_force_cpu= boot parameter.

Link: https://lkml.kernel.org/r/20260118120149.986849-1-pnina.feder@mobileye.com
Reported-by: kernel test robot <lkp@intel.com>
Closes: https://lore.kernel.org/oe-kbuild-all/202601161825.pUmb6FsK-lkp@intel.com/
Signed-off-by: Pnina Feder <pnina.feder@mobileye.com>
Cc: Baoquan He <bhe@redhat.com>
Cc: Petr Mladek <pmladek@suse.com>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 include/linux/panic.h |    8 ++++++++
 1 file changed, 8 insertions(+)

--- a/include/linux/panic.h~panic-add-panic_force_cpu=-parameter-to-redirect-panic-to-a-specific-cpu-fix
+++ a/include/linux/panic.h
@@ -41,6 +41,14 @@ void abort(void);
  * PANIC_CPU_INVALID means no CPU has entered panic() or crash_kexec().
  */
 extern atomic_t panic_cpu;
+
+/*
+ * panic_redirect_cpu is used when panic is redirected to a specific CPU via
+ * the panic_force_cpu= boot parameter. It holds the CPU number that originally
+ * triggered the panic before redirection. A value of PANIC_CPU_INVALID means
+ * no redirection has occurred.
+ */
+extern atomic_t panic_redirect_cpu;
 #define PANIC_CPU_INVALID	-1
 
 bool panic_try_start(void);
_
